src/app.py

```diff
@@ -0,0 +1,57 @@
+# Standard Library
+from logging import getLogger
+
+# Third Party
+from fastapi import FastAPI
+from uvicorn import Config, Server
+
+# Local
+from src.common import settings
+from src.common.utils import setup_logging
+from src.database.handler import DatabaseHandler
+from src.message_queue.handler import MessageQueueHandler
+from src.routers.documents import router as documents_router
+from src.routers.root import router as root_router
+
+setup_logging()
+logger = getLogger(__name__)
+
+# new FastAPI instance
+app = FastAPI()
+
+# link FastAPI routers
+app.include_router(root_router)
+app.include_router(documents_router, prefix="/documents")
+
+
+@app.on_event("startup")
+async def startup():
+    # create database handler
+    database_handler = DatabaseHandler()
+
+    # create message queue handler
+    message_queue_handler = MessageQueueHandler()
+
+
+@app.on_event("shutdown")
+async def shutdown():
+    # close message queue handler
+    database_handler.close()
+
+    # close message queue handler
+    message_queue_handler.close()
+
+
+# custom uvicorn instance to extract the event loop
+def start_uvicorn():
```

Comment:
We use to have it like that in the past as well, however then we found out more convenient to have it directly in the script to be able easily change the server configuration via env. variables and to have server configuration part of the code. It enabled us to run the local server by just running the python script instead of running the uvicorn with some set of parameters. I agree that there are better ways, it is just sometimes it is better to not strictly follow the rules or best practices if the team benefits more from something that is not perfect.

Reply:
Thanks for the explatnation